### DISCUSSION TITLE
Proposal: valid XHTML for basic theme output.

### DISCUSSION THREAD

**Discussion by jayaddison** [sphinx-doc/sphinx#12165] [Feature requests and ideas]
**Is your feature request related to a problem? Please describe.**
It seems that _most_ of the HTML output generated when building Sphinx's own documentation using the `basic` theme (_not_ currently what we [have configured](https://github.com/sphinx-doc/sphinx/blob/d59b158371fc3e477a5ce8dbe9f9500bc962aeeb/doc/conf.py#L34)) is XHTML-compliant - that is, it's also valid XML.

In fact, of `146` HTML files, `144` are valid XML.  The only two exceptions currently are: `search.html` and `domains.html`.

  * On `search.html`, there is a `defer` attribute that is not valid XML.
  * In `domains.html`, there are unclosed `<span>` tags that are not valid XML.

I ran the following utility script to determine that, after using `sphinx-build -b html doc _build` to self-build the documentation -- with the `html_theme` set to `basic`.

```py
from pathlib import Path
from xml.etree.ElementTree import ElementTree

element_tree = ElementTree()
valid = 0
total = 0

build_dir = Path('_build')
for path in sorted(build_dir.rglob('**/*.html')):
    print(path)
    try:
        element_tree.parse(path)
        valid += 1
    except Exception as e:
        print(str(e))
    total += 1

print(f'Valid X(HT)ML : {valid}')
print(f'Total files   : {total}')
```

Trimmed output:

```sh
$ python check.py
...
_build/usage/restructuredtext/domains.html
mismatched tag: line 61, column 2
_build/usage/restructuredtext/field-lists.html
_build/usage/restructuredtext/index.html
_build/usage/restructuredtext/roles.html
_build/usage/theming.html
Valid X(HT)ML : 144
Total files   : 146
```

**Describe the solution you'd like**

 * Output valid XHTML for all pages when using the `basic` theme -- in other words, fix the output for `search.html` and `domains.html`.
 * Add some test coverage -- perhaps to the `html5` builder tests -- to check some simple cases for valid XML. 

**Describe alternatives you've considered**

**Additional context**
I think this would enable us to drop permissive HTML parsers such as `html5lib` and `lxml` from our `sphinx[test]` dependencies.

- Discovered during review-of / experimentation-with #12157.

**Comment by chrisjsewell** [comment]
Sounds reasonable to me 👍 

I guess,from a"theoretical" perspective, I'm interested to know does anyone think there is ever a valid reason to have non-compliant XHTML? Or would we loosely ascribe this to "bugs" in these templates?
